WallsAndHoles/tile.cpp

```diff
@@ -10,7 +10,20 @@ Tile::Tile(const SharedTileTemplate tileTemplate,
     , mYPos(yPos)
     , mRelativeThickness(0)
     , mRelativeHeight(0)
-    , mRelativePosition(QVector2D()) {}
+    , mRelativePosition(QVector2D())
+{
+    connect(tileTemplate.data(), &TileTemplate::exclusivePropertyChanged,
+            this, [this]{ emit tileChanged(mXPos, mYPos); });
+    connect(tileTemplate.data(), &TileTemplate::thicknessChanged,
+            this, &Tile::templateThicknessChanged);
+    connect(tileTemplate.data(), &TileTemplate::positionChanged,
+            this, &Tile::templatePositionChanged);
+}
+
+Tile::~Tile()
+{
+    mTileTemplate->disconnect(this);
```

Comment:
"A signal-slot connection is removed when either of the objects involved are destroyed."  http://doc.qt.io/qt-5/qobject.html#disconnect-2

So this line is not necessary.